src/security/core/tests/common/access_control_wrapper.c

```diff
@@ -63,8 +73,75 @@ struct dds_security_access_control_impl {
   dds_security_access_control *instance;
   enum ac_plugin_mode mode;
   uint32_t not_allowed_mask;
+  ddsrt_mutex_t returns_log_lock;
+  struct ddsrt_circlist returns_log;
+  bool invalid_return;
 };
 
+
+static void init_returns_log(struct dds_security_access_control_impl *impl)
```

Comment:
Done in 7da139a, and I've removed the specific test case for returns-checking as this is now checked in other tests that use the access control test wrapper in wrapper mode. This also revealed a bug in `is_topic_discovery_protected` where the return function call was missing for the topic attributes (which was not really an issue because the return implementation is currently a no-op).